samples/snippets/src/test/java/com/beta/automl/DeleteDatasetTest.java

```diff
@@ -0,0 +1,96 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.beta.automl;
+
+import static com.google.common.truth.Truth.assertThat;
+import static junit.framework.TestCase.assertNotNull;
+
+import com.google.cloud.automl.v1beta1.AutoMlClient;
+import com.google.cloud.automl.v1beta1.Dataset;
+import com.google.cloud.automl.v1beta1.LocationName;
+import com.google.cloud.automl.v1beta1.TextExtractionDatasetMetadata;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+@SuppressWarnings("checkstyle:abbreviationaswordinname")
+public class DeleteDatasetTest {
+
+  private static final String PROJECT_ID = System.getenv("AUTOML_PROJECT_ID");
+  private ByteArrayOutputStream bout;
+  private PrintStream out;
+  private String datasetId;
+
+  private static void requireEnvVar(String varName) {
+    assertNotNull(
+        System.getenv(varName),
+        "Environment variable '%s' is required to perform these tests.".format(varName));
+  }
+
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("AUTOML_PROJECT_ID");
+  }
+
+  @Before
+  public void setUp() throws IOException {
+    // Create a fake dataset to be deleted
+    // Create a random dataset name with a length of 32 characters (max allowed by AutoML)
+    // To prevent name collisions when running tests in multiple java versions at once.
+    // AutoML doesn't allow "-", but accepts "_"
+    String datasetName =
+        String.format("test_%s", UUID.randomUUID().toString().replace("-", "_").substring(0, 26));
+    try (AutoMlClient client = AutoMlClient.create()) {
+      LocationName projectLocation = LocationName.of(PROJECT_ID, "us-central1");
+      TextExtractionDatasetMetadata metadata = TextExtractionDatasetMetadata.newBuilder().build();
+      Dataset dataset =
+          Dataset.newBuilder()
+              .setDisplayName(datasetName)
+              .setTextExtractionDatasetMetadata(metadata)
+              .build();
+      Dataset createdDataset = client.createDataset(projectLocation, dataset);
+      String[] names = createdDataset.getName().split("/");
+      datasetId = names[names.length - 1];
+    }
+
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
+    System.setOut(out);
```

Comment:
Save this and restore it in `tearDown()`

Reply:
Just wanted to share that we updated all the BigQuery Java samples to restore the old/actual out. Here is an example: https://github.com/googleapis/java-bigquery/blob/master/samples/snippets/src/test/java/com/example/bigquery/CreateModelIT.java